sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,210 @@
+        return this.targetThroughput;
+    }
+
+    /**
+     * Set the throughput control group target throughput. The value should be larger than 0.
+     *
+     * @param targetThroughput the target throughput for the throughput control group.
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup targetThroughput(Integer targetThroughput) {
+        checkArgument(
+            targetThroughput != null && targetThroughput > 0,
+            "Target throughput should not be null and should be larger than 0");
+        this.targetThroughput = targetThroughput;
+        return this;
+    }
+
+    /**
+     * Get the throughput control group target throughput threshold.
+     *
+     * @return the target throughput threshold.
+     */
+    public Double getTargetThroughputThreshold() {
+        return this.targetThroughputThreshold;
+    }
+
+    /**
+     * Set throughput control group target throughput threshold. The value should be larger than 0.
+     *
+     * @param targetThroughputThreshold the target throughput threshold for the throughput control group.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup targetThroughputThreshold(Double targetThroughputThreshold) {
+        checkArgument(
+            targetThroughputThreshold != null && targetThroughputThreshold > 0,
+            "Target throughput threshold should not be null and should be larger than 0");
+
+        this.targetThroughputThreshold = targetThroughputThreshold;
+        return this;
+    }
+
+    /**
+     * Get whether this throughput control group will be used by default.
+     *
+     * By default, it is false.
+     *
+     * @return {@code true} this throughput control group will be used by default unless being override. {@code false} otherwise.
+     */
+    public boolean isUseByDefault() {
+        return this.useByDefault;
+    }
+
+    /**
+     * Set the throughput control group to be used by default.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup useByDefault() {
+        this.useByDefault = Boolean.TRUE;
+        return this;
+    }
+
+    /**
+     * Validate whether the throughput control group config is valid.
+     */
+    public void validate() {
+        if (StringUtils.isEmpty(this.groupName)) {
+            throw new IllegalArgumentException("Group name can not be null or empty");
+        }
+        if (this.targetContainer == null) {
+            throw new IllegalArgumentException(String.format("Target container is missing for group %s", this.groupName));
+        }
+        if (this.targetThroughputThreshold == null && this.targetThroughput == null) {
+            throw new IllegalArgumentException("Neither target throughput nor target throughput threshold is defined.");
+        }
+    }
+
+    public String getId() {
+        return String.format(
+            "%s-%s-%s",
```

Comment:
"-" can already be part of the database/collection resource ID; may be a different separator can be used here?

Reply:
Good point, thanks Milis, I will change in next iteration

Reply:
Changed to use "." instead of "-"